core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -236,13 +236,17 @@ class LogCleanerManagerTest extends JUnitSuite with Logging {
       recoveryPoint = 0L,
       scheduler = time.scheduler,
       time = time,
-      brokerTopicStats = new BrokerTopicStats)
+      brokerTopicStats = new BrokerTopicStats,
+      maxProducerIdExpirationMs = 60 * 60 * 1000,
+      producerIdExpirationCheckIntervalMs = 10 * 60 * 1000,
+      logDirFailureChannel = null)
```

Comment:
Should we not pass an instance of `LogDirFailureChannel` instead of `null`? It seems a bit odd to pass an invalid value. If we added some validation to the apply method this test would start failing. Same for every other place we currently pass `null`.